protocol.go

```diff
@@ -108,22 +116,25 @@ func (proto *protocol) Serve(conn net.Conn, userData interface{}) error {
 	}
 
 	for {
-		// Parse a request.
-		req := api.Request{}
-		hr := handlerResponse{}
 
-		err := api.ReadMessage(conn, &req)
+		frame, err := api.ReadFrame(conn)
 		if err != nil {
 			// EOF or the client isn't even sending proper JSON,
 			// just kill the connection
 			return err
 		}
+		if frame.Header.Type != api.TypeCommand {
```

Comment:
Why does the proxy accept frames of type Command only? The shim could send  frames of type Stream for sending stdin.